pkg/templates/nstemplatetiers/nstemplatetier_generator_test.go

```diff
@@ -0,0 +1,236 @@
+package nstemplatetiers_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/templates/nstemplatetiers"
+	testnstemplatetiers "github.com/codeready-toolchain/host-operator/test/templates/nstemplatetiers"
+	testsupport "github.com/codeready-toolchain/toolchain-common/pkg/test"
+
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func TestCreateOrUpdateResources(t *testing.T) {
+
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+
+	t.Run("ok", func(t *testing.T) {
+
+		t.Run("create only", func(t *testing.T) {
+			// given
+			namespace := "host-operator" + uuid.NewV4().String()[:7]
+			client := testsupport.NewFakeClient(t)
+			// set the 'resourceVersion' when creating the object
+			client.MockCreate = func(ctx context.Context, obj runtime.Object) error {
+				if obj, ok := obj.(*toolchainv1alpha1.NSTemplateTier); ok {
+					if obj.ObjectMeta.ResourceVersion != "" {
+						return errors.Errorf("'resourceVersion' field must be specified by the server during object creation")
+					}
+					obj.ObjectMeta.ResourceVersion = "foo-" + obj.ObjectMeta.Name
```

Comment:
I've changed the test to increment the `Generation` instead of changing the `Revision`, while trying to reproduce what the "real" server does.

Reply:
> BTW, why you didn't have to do it in case of creating the inner objects of namespaces in the member operator? https://github.com/codeready-toolchain/member-operator/blob/master/pkg/template/processor.go#L60-L80

I'm not sure I understand what you mean?

Reply:
@MatousJobanek is it better with https://github.com/codeready-toolchain/host-operator/pull/78/commits/e88f525cca19474f5da64795f80f59951808a27e ? and yes, at least on Minishift, I found that it's necessary to set the ResourceVersion from the object returned by `client.Get()` into the object to pass to `client.Update()` otherwise the server would respond with an error, as I commented in the code. I'll see if we need to do the same in https://github.com/codeready-toolchain/member-operator/blob/master/pkg/template/processor.go#L60-L80